packages/filigran-ui/src/components/clients/multi-select.tsx

```diff
@@ -109,154 +163,218 @@ const MultiSelectFormField = React.forwardRef<
       onValueChange(Array.from(selectedValuesSet.current))
     }
 
+    const hiddenCount = selectedValues.length - visibleBadges
+
     return (
-      <Popover
-        open={isPopoverOpen}
-        onOpenChange={setIsPopoverOpen}>
-        <PopoverTrigger asChild>
-          <Button
-            ref={ref}
-            {...props}
-            onClick={() => setIsPopoverOpen(!isPopoverOpen)}
-            className="flex h-auto min-h-9 w-full items-center justify-between rounded border border-input bg-inherit p-1 hover:bg-hover">
-            {selectedValues.length > 0 ? (
-              <div className="flex w-full items-center justify-between">
-                <div className="flex flex-wrap items-center gap-s">
-                  {selectedValues.map((value) => {
-                    const option = options.find((o) => String(o[keyValue]) === value)
+      <TooltipProvider delayDuration={0}>
+        <div
+          className="sr-only"
+          aria-hidden="true">
+          <CloseIcon />
+        </div>
+        <Popover
+          open={isPopoverOpen}
+          onOpenChange={setIsPopoverOpen}>
+          <PopoverTrigger asChild>
+            <Button
+              ref={ref}
+              {...props}
+              onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+              className="flex h-9 w-full items-center justify-between rounded border border-input bg-inherit p-1 hover:bg-hover">
+              {selectedValues.length > 0 ? (
+                <div className="flex w-full items-center">
+                  <div
+                    ref={badgesContainerRef}
+                    className="flex flex-1 items-center gap-s overflow-hidden">
+                    <div
+                      ref={measurementRef}
+                      className="absolute invisible flex items-center gap-s"
+                      aria-hidden="true">
+                      {selectedValues.map((value) => {
+                        const option = options.find(
+                          (opt) => String(opt[keyValue]) === value
+                        )
+                        return (
+                          <Badge key={value}>
+                            {option ? String(option[keyLabel]) : value}
+                            <CloseIcon className="ml-s h-3 w-3" />
+                          </Badge>
+                        )
+                      })}
```

Comment:
We can put in a `useMemo` because otherwise we're doing a find in a map (O(n2) complexity) and it could impact performances